api/graphql/mutations/sample.py

```diff
@@ -36,3 +63,70 @@ async def add_comment(
             content=content,
         )
         return GraphQLComment.from_internal(result)
+
+    # region CREATES
+
+    @strawberry.mutation
+    async def create_sample(
+        self,
+        sample: SampleUpsertInput,
+        info: Info[GraphQLContext, 'SampleMutations'],
+        root: 'ProjectMutations',
+    ) -> str | None:
+        """Creates a new sample, and returns the internal sample ID"""
+        connection: Connection = info.context['connection']
+        st = SampleLayer(connection)
+
+        sample_upsert = SampleUpsert.from_dict(strawberry.asdict(sample))
+        internal_sid = await st.upsert_sample(
+            sample_upsert.to_internal(), project=root.project_id
+        )
+
+        if internal_sid.id:
+            return sample_id_format(internal_sid.id)
+        return None
+
+    @strawberry.mutation
+    async def upsert_samples(
+        self,
+        samples: list[SampleUpsertInput],
+        info: Info[GraphQLContext, 'SampleMutations'],
+        root: 'ProjectMutations',
+    ) -> list[SampleUpsertType] | None:
+        """
+        Upserts a list of samples with sequencing-groups,
+        and returns the list of internal sample IDs
+        """
+        connection: Connection = info.context['connection']
+        st = SampleLayer(connection)
+
+        internal_samples = [
+            SampleUpsert.from_dict(strawberry.asdict(sample)).to_internal()
+            for sample in samples
+        ]
+        upserted = await st.upsert_samples(internal_samples, project=root.project_id)
+
+        return [SampleUpsertType.from_upsert_internal(s) for s in upserted]
+
+    # endregion CREATES
+
+    # region OTHER
+
+    @strawberry.mutation
+    async def update_sample(
+        self,
+        id_: str,
```

Comment:
id is already included in `SampleUpsertInput` it shouldn't be a separate param as well